src/App.tsx

```diff
@@ -15,36 +15,105 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+enum SortType {
+  SORT_FIELD_ALPHABET = 'alphabet',
+  SORT_FIELD_LENGTH = 'length',
+  SORT_FIELD_RESET = '',
+}
+
+function getSortedArray(
+  array: string[],
+  sortField: string,
+  isReversed: boolean,
+): string[] {
+  const preparedGoods = [...array];
+
+  if (sortField) {
+    switch (sortField) {
+      case SortType.SORT_FIELD_ALPHABET:
+        preparedGoods.sort((a, b) => a.localeCompare(b));
+        break;
+      case SortType.SORT_FIELD_LENGTH:
+        preparedGoods.sort((a, b) => a.length - b.length);
+        break;
+      default:
+        break;
+    }
+  }
+
+  if (isReversed) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState<string>('');
+  const [isReversed, setReverseField] = useState<boolean>(false);
+  const visibleGoods = getSortedArray(goodsFromServer, sortField, isReversed);
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={
+            sortField === SortType.SORT_FIELD_ALPHABET
+              ? `button is-info`
+              : `button is-info is-light`
+          }
```

Comment:
pls use [classnames library](https://www.npmjs.com/package/classnames) for all such cases in the project